plugin/pkg/admission/priority/admission.go

```diff
@@ -211,12 +213,17 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 			}
 
 			priority = pc.Value
+
+			if utilfeature.DefaultFeatureGate.Enabled(features.NonPreemptingPriority) {
+				preemptionPolicy = pc.PreemptionPolicy
+			}
 		}
 		// if the pod contained a priority that differs from the one computed from the priority class, error
 		if pod.Spec.Priority != nil && *pod.Spec.Priority != priority {
 			return admission.NewForbidden(a, fmt.Errorf("the integer value of priority (%d) must not be provided in pod spec; priority admission controller computed %d from the given PriorityClass name", *pod.Spec.Priority, priority))
 		}
```

Comment:
if the feature gate is enabled, I'd expect us to check if the incoming pod has a non-nil preemption policy that conflicts with the computed one and fail rather than just stomp it. Need to make sure tests cover these cases:
* pods with nil policy succeed
* pods with non-nil policy that matches the computed policy succeed
* pods with non-nil policy that does not match the computed policy are rejected